app/src/main/java/com/vitorpamplona/amethyst/model/User.kt

```diff
@@ -25,66 +15,56 @@ import java.util.regex.Pattern
 
 val lnurlpPattern = Pattern.compile("(?i:http|https):\\/\\/((.+)\\/)*\\.well-known\\/lnurlp\\/(.*)")
 
-class User(val pubkeyHex: String) {
-    var info: UserMetadata? = null
+class User(override val pubkeyHex: String) : UserInterface {
+    override var info: UserMetadata? = null
+    override var latestContactList: ContactListEvent? = null
+    override var latestBookmarkList: BookmarkListEvent? = null
+    override var notes = setOf<Note>()
+    override var reports = mapOf<UserInterface, Set<Note>>()
+    override var latestReportTime: Long = 0
+    override var zaps = mapOf<Note, Note?>()
+    override var relaysBeingUsed = mapOf<String, RelayInfo>()
+    override var privateChatrooms = mapOf<UserInterface, Chatroom>()
+    override var acceptedBadges: AddressableNote? = null
+    override var liveSet: UserLiveSet? = null
 
-    var latestContactList: ContactListEvent? = null
-    var latestBookmarkList: BookmarkListEvent? = null
-
-    var notes = setOf<Note>()
-        private set
-
-    var reports = mapOf<User, Set<Note>>()
-        private set
-
-    var latestReportTime: Long = 0
-
-    var zaps = mapOf<Note, Note?>()
-        private set
```

Comment:
The only counter-side from moving the properties to the interface is that we cannot specify the visibility of the setters. Eg: `Private setters are not allowed for abstract properties`.
<img width="539" alt="Screenshot 2023-03-25 at 17 56 20" src="https://user-images.githubusercontent.com/5256287/227731071-4f5286ba-319e-4f1e-965d-83951692c190.png">

What do you think, @vitorpamplona? Do we want to have methods for those vars? Or it is ok to keep it like this?

Reply:
Not great since there is a reason to force these fields to be private (an update must trigger a livedata refresh) 

Maybe we take a step back and only do the interface for methods that are going to be tested? When can then have Interfaces for parts of the User class that matter for each test case (and hopefully parts of the app as well)

Reply:
What about creating an interface method for those properties as I had before? This way we will have the benefits from both parts.

Reply:
It's hard to judge it without actually using the interface for something useful. The PR feels like we are doing this just for the sake of code styling than for any useful reason. Because of that, the new methods and classes feel like bloat. 

I think it would make more sense to start designing interfaces when test cases need it. If we are testing Zaps, we might need a UserZappingInterface with just the methods to collect, update and show zaps.

Reply:
Yes, you're right, @vitorpamplona. Ok, I will re-think the approach as you are suggesting 🤔